populus/plugin.py

```diff
@@ -12,17 +24,135 @@
     normalize_object_for_json,
 )
 
+from populus.defaults import (
+    USER_JSON_CONFIG_DEFAULTS,
+    USER_JSON_CONFIG_FILENAME,
+)
+
+from populus.config.helpers import (
+    get_user_default_json_config_file_path,
+)
+
+
+from populus.utils.filesystem import (
+    ensure_path_exists,
+)
 
 CACHE_KEY_MTIME = "populus/project/compiled_contracts_mtime"
 CACHE_KEY_CONTRACTS = "populus/project/compiled_contracts"
 
+BASE_DIR = os.path.dirname(__file__)
+ASSETS_DIR = os.path.join(BASE_DIR, 'assets')
+
+
+testing_project_dir = None
+testing_context = None
+testing_user_config_path = None
+#
+# Hooks
+#
+
+
+def get_populus_option(cmdline_option, ini_option, environ_var, early_config, args):
+
+    for arg in args:
+        option, _, val = arg.partition("=")
+        if option == cmdline_option:
+            return val
+    else:
+        if early_config.getini(ini_option):
+            return early_config.getini(ini_option)
+        else:
+            return os.environ.get(environ_var)
+
+
+def pytest_addoption(parser):
+
+    parser.addoption("--populus-user-config", help="populus global user config json file")
+    parser.addini("populus_user_config", "populus global user config json file")
+    parser.addoption("--populus-project", help="populus project root directory")
+    parser.addini("populus_project", "populus project root directory")
+    parser.addoption("--populus-context", help="populus testing context")
+    parser.addini("populus_context", "populus testing context")
+
+
+def pytest_load_initial_conftests(early_config, parser, args):
+
+    global testing_user_config_path
+    testing_user_config_path = get_populus_option(
+        cmdline_option="--populus-user-config",
+        ini_option="populus_user_config",
+        environ_var="POPULUS_USER_CONFIG",
+        early_config=early_config,
+        args=args
+    )
+    if not testing_user_config_path:
+        testing_user_config_path = get_user_default_json_config_file_path()
+
+    global testing_context
```

Comment:
globals *bad*

It looks like you should be able to use [`pytest_configure`](https://docs.pytest.org/en/latest/writing_plugins.html?highlight=early_config#_pytest.hookspec.pytest_configure) to do everything you're doing here.

Reply:
tried. can try again

Reply:
See https://github.com/Aviah/populus/pull/12